python/tk_framework_desktopserver/server_protocol.py

```diff
@@ -129,56 +181,156 @@ def onMessage(self, payload, is_binary):
         origin_network = self._origin.lower()
         host_network = self.factory.host.lower()
 
         if self._protocol_version == 2:
-
             # Version 2 of the protocol can only answer requests from the site and user the server
             # is authenticated into. Validate this.
 
             # Try to get the user information. If that fails, we need to report the error.
             try:
                 user_id = message["command"]["data"]["user"]["entity"]["id"]
-            except Exception:
-                logger.exception("Unexpected error while trying to retrieve the user id.")
-                self.sendClose(3000, u"No user information was found in this request.")
+            except Exception as e:
+                logger.exception("Unexpected error while trying to retrieve the user id:")
+                self.sendClose(*self.USER_INFO_NOT_FOUND)
                 return
         else:
             user_id = None
 
-        # If the hosts are different or the user ids are different, report an error.
-        if host_network != origin_network or (user_id is not None and user_id != self.factory.user_id):
-            logger.debug("Browser integration request received a different user.")
-            logger.debug("Desktop site: %s", host_network)
-            logger.debug("Desktop user: %s", self.factory.user_id)
-            logger.debug("Origin site: %s", origin_network)
-            logger.debug("Origin user: %s", user_id)
+        if not self._validate_user(host_network, user_id, origin_network):
             self.factory.notifier.different_user_requested.emit(self._origin, user_id)
-            self.sendClose(
-                3001,
-                u"You are not authorized to make browser integration requests. "
-                u"Please re-authenticate in your desktop application."
-            )
+            self.sendClose(*self.UNAUTHORIZED_USER)
             return
 
-        # Run each request from a thread, even though it might be something very simple like opening a file. This
-        # will ensure the server is as responsive as possible. Twisted will take care of the thread.
+        # Run each request from a thread, even though it might be something very simple like opening
+        # a file. This will ensure the server is as responsive as possible. Twisted will take care
+        # of the thread.
         reactor.callInThread(
             self._process_message,
             message_host,
             message,
             message["protocol_version"],
         )
 
+    def _validate_user(self, host_network, user_id, origin_network):
+        """
+        Validates if the user from the browser can connect to this server.
+
+        :returns: True if the user can connect, False otherwise.
+        """
+        # If the hosts are different or the user ids are different, report an error.
+        if host_network != origin_network or (user_id is not None and user_id != self.factory.user_id):
```

Comment:
[minor] My feeble, 10pm brain has trouble parsing this condition because of all of the nots. Any chance you could flip it around and set the if condition up to return True and return False if it fails the validation checks as the else statement?

Reply:
Good point. Done.